ext/spl/php_spl.c

```diff
@@ -154,8 +154,8 @@ PHP_FUNCTION(class_uses)
 		return;
 	}
 	if (Z_TYPE_P(obj) != IS_OBJECT && Z_TYPE_P(obj) != IS_STRING) {
-		php_error_docref(NULL, E_WARNING, "object or string expected");
-		RETURN_FALSE;
+		zend_type_error("object or string expected");
+		return;
```

Comment:
The changes in this file are okay (pls commit), but the changes to SplFileObject should imho only be done if we have modified normal f* functions in the same way, which doesn't seem to be the case.